providers/enchant_hspell.c

```diff
@@ -86,8 +86,13 @@ static int
 hspell_dict_check (EnchantDict * me, const char *const word, size_t len)
 {
 	struct dict_radix *hspell_dict = (struct dict_radix *)me->user_data;
-	char *iso_word = hspell_convert_to_iso8859_8 (me, word, len);
-	g_return_val_if_fail (iso_word, -1);
+	char *iso_word = NULL;
+
+	iso_word = hspell_convert_to_iso8859_8 (me, word, len);
```

Comment:
Please could you just restore the original line 89, i.e. merge the declaration and assignment of `iso_word` in each case?